### PR TITLE
[`flake8-bandit`] Add Rule for `S508` (snmp insecure version) & `S509` (snmp weak cryptography)

### PR DESCRIPTION
**Pull request by saadmk11**
ref: https://github.com/charliermarsh/ruff/issues/1646

### CODE REVIEW

File: src/violations.rs [hunk 1]
```diff
@@ -4792,6 +4792,33 @@ impl Violation for UnsafeYAMLLoad {
...
+        "Possible insecure use of SNMPv1 and SNMPv2. You should use SNMPv3 if able".to_string()
+    }
+
+    fn placeholder() -> Self {
+        SnmpInsecureVersion
+    }
+}
+
+define_violation!(
+    pub struct SnmpWeakCriptography;
```
Comment (messense) [inline src/violations.rs:4809]:
```suggestion
    pub struct SnmpWeakCryptography;
```

File: src/violations.rs [hunk 2]
```diff
@@ -4792,6 +4792,33 @@ impl Violation for UnsafeYAMLLoad {
...
+);
+impl Violation for SnmpWeakCriptography {
```
Comment (messense) [inline src/violations.rs:4811]:
```suggestion
impl Violation for SnmpWeakCryptography {
```

File: src/violations.rs [hunk 3]
```diff
@@ -4792,6 +4792,33 @@ impl Violation for UnsafeYAMLLoad {
...
+    fn message(&self) -> String {
+        "Possible use of SNMPv3 without encryption. `noAuthNoPriv` & `authNoPriv` is insecure"
+            .to_string()
+    }
+
+    fn placeholder() -> Self {
+        SnmpWeakCriptography
```
Comment (messense) [inline src/violations.rs:4818]:
```suggestion
        SnmpWeakCryptography
```

File: src/registry.rs [hunk 4]
```diff
@@ -416,6 +416,8 @@ define_rule_mapping!(
     S324 => violations::HashlibInsecureHashFunction,
     S501 => violations::RequestWithNoCertValidation,
     S506 => violations::UnsafeYAMLLoad,
+    S508 => violations::SnmpInsecureVersion,
+    S509 => violations::SnmpWeakCriptography,
```
Comment (messense) [inline src/registry.rs:420]:
```suggestion
    S509 => violations::SnmpWeakCryptography,
```

File: src/flake8_bandit/rules/snmp_weak_cryptography.rs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    import_aliases: &FxHashMap<&str, &str>,
+) -> Option<Diagnostic> {
+    let call_path = dealias_call_path(collect_call_paths(func), import_aliases);
+
+    if match_call_path(&call_path, "pysnmp.hlapi", "UsmUserData", from_imports) {
+        let call_args = SimpleCallArgs::new(args, keywords);
+
+        if call_args.len() < 3 {
+            return Some(Diagnostic::new(
+                violations::SnmpWeakCriptography,
```
Comment (messense) [inline src/flake8_bandit/rules/snmp_weak_cryptography.rs:24]:
```suggestion
                violations::SnmpWeakCryptography,
```

**Review by messense — COMMENTED** [review]
Typo: `SnmpWeakCriptography` should be `SnmpWeakCryptography`

**Comment by saadmk11** [comment]
Making a lot of typo's today 😅

**Pull request merged**